### PR TITLE
Fixes for consul and eureka istio.io tasks

### LINKED ISSUE #2080: BookInfo with Consul: Istio Routing not working
Tested using the 0.3 release with docker on windows.

## Steps taken:
- Used `install/*` and `samples/` folder as of e52148bfcbe0caed864aeeb3f0c937c2ca28866f
  since install/consul lacks the kubeconfig file, which leads to very annoying issues to debug
  (Using just kubeconfig from master shows the same behavior)
- Followed steps on https://istio.io/docs/setup/consul/quick-start.html 
- Followed steps on https://istio.io/docs/guides/bookinfo.html#running-on-docker-with-consul-or-eureka
- Ran `istioctl create -f samples/bookinfo/consul/route-rule-all-v1.yaml`
- Accessed `http://localhost:9081/productpage` mutliple times

The last 2 steps are basically the first 2 from https://istio.io/docs/tasks/traffic-management/request-routing.html

## Expected
The round-robin load balance between the different review-service-versions
stops (e.g. consistent result, which would mean that istio actually handles the traffic)

## Actual result
Sometimes the rating stars are red, sometimes they are missing (as they consistently should),
sometimes they are black.

Any idea what's going wrong here, am I just better of with using minikube for now?

### PR DESCRIPTION
**Pull request by GregHanson**
**What this PR does / why we need it**:
default discovery istio-pilot:<port> was changed from 8080 to 15003, broke consul and eureka docker-compose yamls

**Which issue this PR fixes**: fixes #2080 

**Special notes for your reviewer**:

**Release note**:

```release-note
Fixed issue with Pilot port in Consul and Eureka samples.
```

### CODE REVIEW

**Comment by rshriram** [comment]
@rkpagadala just a note: next time you change any of the install files for kube, please pay attention to install files in Consul/Eureka as well. There is no istio-on-istio for platforms other than kubernetes.

**Comment by kyessenov** [comment]
The post-mortem should be: let's get some automated checks for this :)

**Comment by rshriram** [comment]
Well, that could not work because consul/nomad is a different orchestration platform, while all tests were deeply tied to kubernetes. Its not like we had the flexibility to add other platforms.

**Comment by kyessenov** [comment]
The test should run separately. Perhaps, another circle job that only
exercises consul.

**Comment by rshriram** [comment]
And hence the reason for moving to circleci :). Was not possible with prow,gke, inhouse testing infrastructure.

**Comment by kyessenov** [comment]
Prow is just a bash script running in a container, not much different from
circle. I agree with you, that we should have more cooperation on the
testing front.

**Comment by ijsnellf** [comment]
Related to https://github.com/istio/istio/issues/2168

**Comment by rkpagadala** [comment]
@rshriram understood. Having tests really helps. Thanks

File: samples/bookinfo/eureka/cleanup.sh [hunk 1]
```diff
@@ -36,6 +36,7 @@ done
 export OUTPUT=$(mktemp)
 echo "Application cleanup may take up to one minute"
 docker-compose -f $SCRIPTDIR/bookinfo.yaml down > ${OUTPUT} 2>&1
+docker-compose -f $SCRIPTDIR/bookinfo.sidecar.yaml down > ${OUTPUT} 2>&1
```
Comment (mgtriffid) [inline samples/bookinfo/eureka/cleanup.sh]: I guess it should be "sidecars", not "sidecar", right?

**Comment by mgtriffid** [comment]
Hm, I'm experiencing issue https://github.com/istio/issues/issues/185, and after I added `--discoveryAddress istio-pilot:8080` issue has gone. Is flag actually needed, should it be present in template for generating bookinfo.sidecars.yaml? Or this was a workaround and there is some cleaner way to fix the issue?

**Pull request merged**